### ISSUE TITLE
MD034 false positive and wrong fix

### ISSUE THREAD

**Issue by vkucera** [DavidAnson/markdownlint#615]
Rule MD034 is triggered for an URL appearing between `>` and `<` of the `<a></a>` HTML tag.
There are two problems with this.

1. First, the rule should not be triggered at all, since this should not be considered a bare URL.
2. Second, markdownlint fixes this by putting angle brackets around the first occurrence of the URL, i.e. the actual address inside the `<a` tag instead of putting them around the URL between `>` and `<` which actually triggered the rule.
E.g.:
```html
<a href="https://github.com">https://github.com</a>
```
becomes
```html
<a href="<https://github.com>">https://github.com</a>
```
See [demo](https://dlaa.me/markdownlint/#%25m%3Ca%20href%3D%22https%3A%2F%2Fgithub.com%22%3Ehttps%3A%2F%2Fgithub.com%3C%2Fa%3E%0A)

**Comment by DavidAnson** [comment]
Can I ask why use an HTML `a` tag at all? Why not use the Markdown syntax `<url>` which does the same thing and is shorter and easier to read?

**Comment by vkucera** [comment]
In general, HTML tags are needed for some features not supported by Markdown, e.g. opening a hyperlink in a new tab with `target=_blank` or picture size.
